app/models/user.rb

```diff
@@ -0,0 +1,34 @@
+class User < ActiveRecord::Base
+  has_many :tweets
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
+                                  dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :passive_relationships, class_name: "Relationship",
+                                   foreign_key: "followed_id",
+                                   dependent: :destroy
+  has_many :followers, through: :passive_relationships, source: :follower
+
+  has_secure_password
+
+  def slug
+    self.username.downcase.gsub(" ", "-")
+  end
+
+  def self.find_by_slug(slug)
+      self.all.detect { |instance| instance.username.downcase == slug.gsub("-", " ")}
```

Comment:
Instead of redefining your own `find_by_slug` method, you can utilize the existing ActiveRecord `find_by` method.

```ruby
def self.find_by_slug(slug)
  User.find_by(slug: slug)
end
```
This returns an ActiveRecord::Collection object, allowing you to chain additional ActiveRecord methods.